src/WebJobs.Script.WebHost/App_Start/AutofacBootstrap.cs

```diff
@@ -13,16 +14,15 @@ internal static void Initialize(ScriptSettingsManager settingsManager, Container
         {
             builder.RegisterInstance(settingsManager);
 
-            // register the resolver so that it is disposed when the container
-            // is disposed
-            var webHostResolver = new WebHostResolver(settingsManager);
-            builder.RegisterInstance(webHostResolver);
+            builder.RegisterType<WebHostResolver>().SingleInstance();
 
             // these services are externally owned by the WebHostResolver, and will be disposed
             // when the resolver is disposed
-            builder.Register<ISecretManager>(ct => webHostResolver.GetSecretManager(settings)).ExternallyOwned();
-            builder.Register<WebScriptHostManager>(ct => webHostResolver.GetWebScriptHostManager(settings)).ExternallyOwned();
-            builder.Register<WebHookReceiverManager>(ct => webHostResolver.GetWebHookReceiverManager(settings)).ExternallyOwned();
+            builder.RegisterType<DefaultSecretManagerFactory>().As<ISecretManagerFactory>().SingleInstance();
+            builder.Register<TraceWriter>(ct => ct.ResolveOptional<WebScriptHostManager>()?.Instance?.TraceWriter ?? NullTraceWriter.Instance).ExternallyOwned();
```

Comment:
Not happy with this yet, but this gives us what we need for now. Getting to where we want to be (the broader logging changes we have planned) would address this.